File: week-02/day-02/diagonal_matrix.py
# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

def matrix(num):
    outer = []
    for i in range(num):
        inner = []
        for j in range(num):
            if j == i:
                inner.append(1)
            else:
                inner.append(0)
        outer.append(inner)
    return(outer)
    
print(matrix(4))

File: week-07/day-03/diagonal_matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

var size = 4;

for (let j = 0; j < size; j +=1) {
    console.log('0 '.repeat(size - j - 1) + '1 ' + '0 '.repeat(j))
}